app/controllers/posts_controller.rb

```diff
@@ -3,10 +3,26 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @posts = Post.where(author_id: params[:user_id]).order(id: :asc)
-    @posts = @posts.paginate(page: params[:page], per_page: 2)
+    @posts = @posts.paginate(page: params[:page], per_page: 10)
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author = current_user
+    if @post.save
+      flash[:sucess] = 'Post created successfuly!'
+      redirect_to user_posts_url
+    else
+      flash.now[:error] = 'Error, the post cannot be created!'
+      render :new, locals: { post: @post }
+    end
+  end
 end
```

Comment:
# Optional
- I advise you to add `params` method because it is a better security practice to help prevent accidentally allowing users to update sensitive model attributes.
- [Lesson link about the strong parameters](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters)
👇 
```ruby
 private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

Reply:
✅ Done

Thanks for your suggestion.